sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Scenarios/BufferedProducerTest.cs

```diff
@@ -0,0 +1,92 @@
+
+    /// <summary>The index used to determine which role should be run if this is a distributed test run.</summary>
+    private readonly string _jobIndex;
+
+    /// <summary> The <see cref="Metrics"/> instance used to send metrics to application insights.</summary>
+    private Metrics _metrics;
+
+    /// <summary> The array of <see cref="Role"/>s needed to run this test scenario.</summary>
+    private static Role[] _roles = {Role.BufferedPublisher, Role.BufferedPublisher};
+
+    /// <summary>
+    ///  Initializes a new <see cref="BufferedProducerTest"/> instance.
+    /// </summary>
+    ///
+    /// <param name="testConfiguration">The <see cref="TestConfiguration"/> to use to configure this test run.</param>
+    /// <param name="metrics">The <see cref="Metrics"/> to use to send metrics to Application Insights.</param>
+    /// <param name="jobIndex">An optional index used to determine which role should be run if this is a distributed run.</param>
+    ///
+    public BufferedProducerTest(TestConfiguration testConfiguration, Metrics metrics, string jobIndex = default)
+    {
+        _testConfiguration = testConfiguration;
+        _jobIndex = jobIndex;
+        _metrics = metrics;
+        _metrics.Client.Context.GlobalProperties["TestRunID"] = $"net-buff-prod-{Guid.NewGuid().ToString()}";
+    }
+
+    /// <summary>
+    ///   Runs all of the roles required for this instance of the Buffered Producer test scenario.
+    /// </summary>
+    ///
+    /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+    ///
+    public async Task RunTestAsync(CancellationToken cancellationToken)
+    {
+        var runAllRoles = !int.TryParse(_jobIndex, out var roleIndex);
+        var testRunTasks = new List<Task>();
+
+        if (runAllRoles)
+        {
+            foreach (Role role in _roles)
+            {
+                testRunTasks.Add(RunRoleAsync(role, cancellationToken));
+            }
+        }
+        else
+        {
+            testRunTasks.Add(RunRoleAsync(_roles[roleIndex], cancellationToken));
+        }
+
+        await Task.WhenAll(testRunTasks).ConfigureAwait(false);
+    }
+
+    /// <summary>
+    ///   Creates a role instance and runs that role.
+    /// </summary>
+    ///
+    /// <param name="role">The <see cref="Role"/> to run.</param>
+    /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+    ///
+    public async Task RunRoleAsync(Role role, CancellationToken cancellationToken)
+    {
+        switch (role)
+        {
+            case Role.BufferedPublisher:
+                var publisherConfiguration = new BufferedPublisherConfiguration();
+                var publisher = new BufferedPublisher(_testConfiguration, publisherConfiguration, _metrics);
+                await publisher.StartAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Pragmatically, it probably won't be that drastic in our case, but it can sometimes take a bit for the first yield.   Not a blocker, but worth considering.

Reply:
Ah I see that makes sense. Thank you!